esphome/components/axs15231/touchscreen/axs15231_touchscreen.cpp

```diff
@@ -0,0 +1,63 @@
+#include "axs15231_touchscreen.h"
+
+#include "esphome/core/helpers.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace axs15231 {
+
+static const char *const TAG = "ax15231.touchscreen";
+
+constexpr static const uint8_t AXS_READ_TOUCHPAD[11] = {0xb5, 0xab, 0xa5, 0x5a, 0x0, 0x0, 0x0, 0x8};
+
+#define ERROR_CHECK(err) \
+  if ((err) != i2c::ERROR_OK) { \
+    this->status_set_warning("Failed to communicate"); \
+    return; \
+  }
+
+void AXS15231Touchscreen::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up AXS15231 Touchscreen...");
+  if (this->reset_pin_ != nullptr) {
+    this->reset_pin_->setup();
+    this->reset_pin_->digital_write(false);
+    delay(5);
+    this->reset_pin_->digital_write(true);
+    delay(10);  // NOLINT
+  }
+  if (this->interrupt_pin_ != nullptr) {
+    this->interrupt_pin_->pin_mode(gpio::FLAG_INPUT);
+    this->interrupt_pin_->setup();
+    this->attach_interrupt_(this->interrupt_pin_, gpio::INTERRUPT_FALLING_EDGE);
+  }
+  this->x_raw_max_ = this->display_->get_native_width();
+  this->y_raw_max_ = this->display_->get_native_height();
+  ESP_LOGCONFIG(TAG, "AXS15231 Touchscreen setup complete");
+}
+
+void AXS15231Touchscreen::update_touches() {
+  i2c::ErrorCode err;
+  uint8_t data[8]{};
+
+  err = this->write(AXS_READ_TOUCHPAD, sizeof(AXS_READ_TOUCHPAD), false);
+  ERROR_CHECK(err);
+  err = this->read(data, sizeof(data));
+  ERROR_CHECK(err);
+  if (data[0] != 0)  // no touches
+    return;
+  uint16_t x = encode_uint16(data[2] & 0xF, data[3]);
+  uint16_t y = encode_uint16(data[4] & 0xF, data[5]);
+  this->add_raw_touch_position_(0, x, y);
+}
```

Comment:
Don't forget to clear previous warnings if it made it this far.
```suggestion
  this->status_clear_warning();
}
```